libbeat/processors/fingerprint/fingerprint_test.go

```diff
@@ -0,0 +1,385 @@
+				"fields": []string{"timestamp"},
+			})
+			assert.NoError(t, err)
+
+			p, err := New(testConfig)
+			assert.NoError(t, err)
+
+			testEvent := &beat.Event{
+				Fields: test.event,
+			}
+			newEvent, err := p.Run(testEvent)
+			assert.NoError(t, err)
+
+			v, err := newEvent.GetValue("fingerprint")
+			assert.NoError(t, err)
+			assert.Equal(t, expectedFingerprint, v)
+		})
+	}
+}
+
+func TestTargetField(t *testing.T) {
+	testFields := common.MapStr{
+		"field1": "foo",
+		"nested": common.MapStr{
+			"field": "bar",
+		},
+		"unused_field": "baz",
+	}
+	expectedFingerprint := "4cf8b768ad20266c348d63a6d1ff5d6f6f9ed0f59f5c68ae031b78e3e04c5144"
+
+	tests := map[string]struct {
+		targetField string
+	}{
+		"root":   {"target_field"},
+		"nested": {"nested.target_field"},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			testConfig, err := common.NewConfigFrom(common.MapStr{
+				"fields":       []string{"field1"},
+				"target_field": test.targetField,
+			})
+			assert.NoError(t, err)
+
+			p, err := New(testConfig)
+			assert.NoError(t, err)
+
+			testEvent := &beat.Event{
+				Fields:    testFields.Clone(),
+				Timestamp: time.Now(),
+			}
+			newEvent, err := p.Run(testEvent)
+			assert.NoError(t, err)
+
+			v, err := newEvent.GetValue(test.targetField)
+			assert.NoError(t, err)
+			assert.Equal(t, expectedFingerprint, v)
+
+			_, err = newEvent.GetValue("fingerprint")
+			assert.EqualError(t, err, common.ErrKeyNotFound.Error())
+		})
+	}
+}
+
+func TestSourceFieldErrors(t *testing.T) {
+	testFields := common.MapStr{
+		"field1": "foo",
+		"field2": "bar",
+		"complex_field": map[string]interface{}{
+			"child": "qux",
+		},
+		"unused_field": "baz",
+	}
+
+	tests := map[string]struct {
+		fields         []string
+		expectedErrMsg string
+	}{
+		"missing": {
+			[]string{"field1", "missing_field"},
+			"failed to compute fingerprint: failed to find field [missing_field] in event: key not found",
+		},
+		"non-scalar": {
+			[]string{"field1", "complex_field"},
+			"failed to compute fingerprint: cannot compute fingerprint using non-scalar field [complex_field]",
+		},
+	}
+
+	for name, test := range tests {
+		t.Run(name, func(t *testing.T) {
+			testConfig, err := common.NewConfigFrom(common.MapStr{
+				"fields": test.fields,
+				"method": "sha256",
+			})
+			assert.NoError(t, err)
+
+			p, err := New(testConfig)
+			assert.NoError(t, err)
+
+			testEvent := &beat.Event{
+				Fields:    testFields.Clone(),
+				Timestamp: time.Now(),
+			}
+			_, err = p.Run(testEvent)
+			assert.EqualError(t, err, test.expectedErrMsg)
+		})
+	}
+}
+
+func TestInvalidConfig(t *testing.T) {
+	tests := map[string]struct {
+		config         common.MapStr
+		expectedErrMsg string
+	}{
+		"no fields": {
+			common.MapStr{
+				"fields": []string{},
+				"method": "sha256",
+			},
+			"failed to unpack fingerprint processor configuration: empty field accessing 'fields'",
```

Comment:
nit; it's ok for now. But instead of checking strings one should consider typed errors that also carries some context.

Reply:
Implemented typed errors in 49a0e4f7286d22d2bcc121ac23fff06ce52edd09 and 65627787e402421c7657a53f0d7489792c3675f8.